mycroft/util/audio_utils.py

```diff
@@ -0,0 +1,199 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Contains simple tools for performing audio related tasks such as playback
+of audio, recording and listing devices.
+"""
+from copy import deepcopy
+import os
+import pyaudio
+import re
+import subprocess
+
+import mycroft.configuration
+
+from .string_utils import get_http
+from .log import LOG
+
+
+def play_audio_file(uri: str, environment=None):
+    """Play an audio file.
+
+    This wraps the other play_* functions, choosing the correct one based on
+    the file extension. The function will return directly and play the file
+    in the background.
+
+    Arguments:
+        uri:    uri to play
+        environment (dict): optional environment for the subprocess call
+
+    Returns: subprocess.Popen object. None if the format is not supported or
+             an error occurs playing the file.
+    """
+    extension_to_function = {
+        '.wav': play_wav,
+        '.mp3': play_mp3,
+        '.ogg': play_ogg
+    }
+    _, extension = os.path.splitext(uri)
+    play_function = extension_to_function.get(extension.lower())
+    if play_function:
+        return play_function(uri, environment)
+    else:
+        LOG.error("Could not find a function capable of playing {uri}."
+                  " Supported formats are {keys}."
+                  .format(uri=uri, keys=list(extension_to_function.keys())))
+        return None
+
+
+_ENVIRONMENT = deepcopy(os.environ)
+_ENVIRONMENT['PULSE_PROP'] = 'media.role=music'
+
+
+def _get_pulse_environment(config):
+    """Return environment for pulse audio depeding on ducking config."""
+    tts_config = config.get('tts', {})
+    if tts_config and tts_config.get('pulse_duck'):
+        return _ENVIRONMENT
+    else:
+        return os.environ
+
+
+def play_wav(uri, environment=None):
+    """Play a wav-file.
+
+    This will use the application specified in the mycroft config
+    and play the uri passed as argument. The function will return directly
+    and play the file in the background.
+
+    Arguments:
+        uri:    uri to play
+        environment (dict): optional environment for the subprocess call
+
+    Returns: subprocess.Popen object
+    """
+    config = mycroft.configuration.Configuration.get()
+    environment = environment or _get_pulse_environment(config)
+    play_cmd = config.get("play_wav_cmdline")
+    play_wav_cmd = str(play_cmd).split(" ")
+    for index, cmd in enumerate(play_wav_cmd):
+        if cmd == "%1":
+            play_wav_cmd[index] = (get_http(uri))
+    try:
+        return subprocess.Popen(play_wav_cmd, env=environment)
+    except Exception as e:
+        LOG.error("Failed to launch WAV: {}".format(play_wav_cmd))
+        LOG.debug("Error: {}".format(repr(e)), exc_info=True)
```

Comment:
Most of the time in this case the exception information confuses more than it helped so a simpler error message with the most relevant information was added in it's place we didn't want to remove the possibility to get the entire stacktrace logged completely so it was set as debug only.

Reply:
What was confusing about the exception information?  The only difference between this code and a log.exception() call is that the traceback appears in a different log message.  Making a user turn on debug mode and recreate the issue just to get the stack trace seems counter-intuitive to me.

Reply:
It was originally a plain exception but was a bit verbose and added very little useful information so it was reduced to this a couple of years ago making the default info less verbose while keeping the option to have the backtrace for the rare occations where it's of use.

Reply:
I don't agree with verbosity as a rationale for omitting a traceback from an error message.  Consistency in how this is done is more important.  One-off differences like this make very little sense to me.  Resolving because I am done fighting it.